composer/datasets/in_context_learning_evaluation.py

```diff
@@ -224,275 +283,220 @@ def _prep_examples(self,
-        return len(self.encoded_dataset)
+        prompt_and_fewshot = self.generate_few_shot_text(num_fewshot, example_idx, prompt_string, fewshot_rng)
+        ctxt = self.construct_context(example, prompt_and_fewshot, add_answer=False)
+        tokenized_example = self.tokenize_example(prompt_and_fewshot, ctxt, example)
+        return tokenized_example
 
     def collate_fn(self, data):
-        inputs, answers = [], []
-        cot_delimiter = ''
-
-        for sample in data:
-            preamble, context, aliases = (sample['preamble'], sample['context'], sample['aliases'])
-            context_enc = preamble['input_ids'] + context['input_ids']
-            inp, _ = _make_padded_input(context_enc, [],
-                                        self.max_seq_len - self.max_answer_length,
-                                        self.pad_tok_id,
-                                        padding_side=self.padding_side)
-
-            inputs.append(inp)
-            answers.append(aliases)
-
-            # We will search for the answer within the portion of the model response
-            # beginning with `cot_delimiter`
-            cot_delimiter = sample['cot_delimiter']
-
         batch = {
-            'input_ids': torch.stack(inputs),
-            'mode': 'generate',
-            'labels': answers,
-            'cot_delimiter': cot_delimiter,
-            'generation_length': self.max_answer_length,
-            'generation_kwargs': {
-                'pad_token_id': self.pad_tok_id,
-                'use_cache': True
-            }
+            'input_ids': [],
+            'continuation_indices': [],
+            'mode': 'icl_task',
+            'labels': [],
         }
+        for data_pair in data:
+            context_enc = data_pair['preamble']['input_ids'] + data_pair[self.context_key]['input_ids']
+
+            inp, continuation_span = _make_padded_input(context_enc, data_pair['continuation']['input_ids'],
+                                                        self.max_seq_len, self.pad_tok_id)
 
+            batch['input_ids'].append(inp)
+            batch['continuate_indicies'].append(continuation_span)
+            batch['labels'].append(inp)
+
+        batch = {k: torch.stack(v) if k in self.stacked_keys else v for k, v in batch.items()}
         batch['attention_mask'] = ~(batch['input_ids'] == self.pad_tok_id)
         return batch
 
-    def get_num_samples_in_batch(self, batch) -> int:
-        return batch['input_ids'].shape[0]
-
     def split_batch(self, batch: Any, microbatch_size: int):
         # Don't split kwargs that don't change
         # Normally split torch tensors
         # List split lists of strings
-        no_split = ['mode', 'generation_length', 'generation_kwargs', 'cot_delimiter']
-        normal_split = ['input_ids', 'attention_mask']
-        list_split = ['labels']
         chunked = {}
         for k, v in batch.items():
-            if k in no_split:
+            if k in self.dont_split_keys:
```

Comment:
are these attrs defined for the base class somewhere?

Reply:
added. I chose not to bcus they were different and each class defined them within their class, but better to just pass them in.